### PR TITLE
replace u8 direction field with Direction type

### PR DESCRIPTION
**Pull request by DoTheBestToGetTheBest**
according to https://github.com/ethereum-optimism/optimism/blob/40750a58e7a4a6f06370d18dfe6c6eab309012d9/op-node/p2p/rpc_api.go#L26C24-L26C38

implementation reference https://github.com/libp2p/go-libp2p/blob/d55bed5f78fad923353b92c7d1db64c758bcbf9c/core/network/network.go#L26

### CODE REVIEW

File: crates/rpc-types/src/net.rs [hunk 1]
```diff
@@ -149,7 +149,42 @@ impl From<u8> for Connectedness {
...
+#[repr(u8)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize, Default)]
+pub enum Direction {
+    /// DirUnknown is the default direction when the direction is not specified.
+    #[default]
+    DirUnknown = 0,
+    /// DirInbound is for when the remote peer initiated the connection.
+    DirInbound = 1,
+    /// DirOutbound is for when the local peer initiated the connection.
+    DirOutbound = 2,
```
Comment (mattsse) [inline crates/rpc-types/src/net.rs:162]: lets drop the Dir prefix

File: crates/rpc-types/src/net.rs [hunk 2]
```diff
@@ -149,7 +149,42 @@ impl From<u8> for Connectedness {
         }
     }
 }
+/// Direction represents the direction of a connection.
+#[repr(u8)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize, Default)]
```
Comment (mattsse) [inline crates/rpc-types/src/net.rs:154]: does this serde as number?

please add a test

File: crates/rpc-types/src/net.rs [hunk 3]
```diff
@@ -149,7 +149,42 @@ impl From<u8> for Connectedness {
...
+impl From<u8> for Direction {
+    fn from(num: u8) -> Self {
+        match num {
+            0 => Direction::DirInbound,
+            1 => Direction::DirOutbound,
+            2 => Direction::DirUnknown,
+            _ => Direction::DirInbound,
+        }
+    }
+}
```
Comment (mattsse) [inline crates/rpc-types/src/net.rs:173]: don't think we can do this, mapping unsupported numbers to inbound doesn't feel right, please convert to tryfrom

**Review by mattsse — CHANGES_REQUESTED** [review]

**Comment by DoTheBestToGetTheBest** [comment]
done  @mattsse thanks

**Review by mattsse — APPROVED** [review]

**Pull request merged**